### PR TITLE
Assignment 5

### PR DESCRIPTION
**Pull request by CDobbins1**

### CODE REVIEW

File: Mailbox/Mailboxes.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿using System;
 using System.Collections.Generic;
-
+#pragma warning disable CA1710
```
Comment (Fervidusletum) [inline Mailbox/Mailboxes.cs]: Dont forget to add a comment justifying why you're disabling a warning.

File: Mailbox/Mailbox.cs [hunk 2]
```diff
@@ -1,4 +1,23 @@
...
+    public class Mailbox
+    {
+        public Sizes Size { get; }
+        public (int, int) Location { get; }
+        public Person Owner { get; }
 
+        public Mailbox(Sizes size, (int, int) location, Person owner)
+        {
+            Size = size;
+            Location = location;
```
Comment (Fervidusletum) [inline Mailbox/Mailbox.cs]: Should have a check to make sure the location is valid for `Mailboxes` class.  Would be a good idea to verify size also.  For example, you dont want a `Sizes.Large|Sizes.Medium`.  I could see reasons for allowing owner to be null.

File: Mailbox/Program.cs [hunk 3]
```diff
@@ -89,17 +90,57 @@ static void Main(string[] args)
...
+            var sb = new StringBuilder();
+            foreach (Mailbox mb in mailboxes)
+            {
+                if (!hs.Contains(mb.Owner))
+                {
+                    sb.AppendLine(mb.Owner.ToString());
+                    hs.Add(mb.Owner);
+                }
+            }
+            return sb.ToString();
```
Comment (Fervidusletum) [inline Mailbox/Program.cs]:
```c#
            var hs = new HashSet<Person>();
            foreach (Mailbox mb in mailboxes)
            {
                 hs.Add(mb.Owner);
            }
            return string.Join(Environment.NewLine, hs);
```
Bit of a cleaner option for concatenating and returning all unique owners.

File: Mailbox.Tests/PersonTests.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Mailbox.Tests
+{
+    [TestClass]
+    public class PersonTests
```
Comment (Fervidusletum) [inline Mailbox.Tests/PersonTests.cs]: Couldn't hurt to do a few more edge case checks, like `person.Equals(null)`, and maybe a simple check just to ensure the operators `==` and `!=` are working as expected.

File: Mailbox/Mailbox.cs [hunk 5]
```diff
@@ -1,4 +1,23 @@
...
+            Owner = owner;
+        }
+
+        public override string ToString()
+        {
+            string outputSize = Size != 0 ? Size.ToString() : "";
```
Comment (Keboo) [inline Mailbox/Mailbox.cs]: The expectation was for the size to include " Premium" when that flag was set, however I realize that bullet point was really poorly worded.
